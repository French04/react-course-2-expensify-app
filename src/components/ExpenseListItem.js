import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

numeral.register('locale', 'it', {
    delimiters: {
        thousands: '.',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
        ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '€'
    }
});
numeral.locale('it');

const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <Link className="page-expense__item" to={`/edit/${id}`}>
        <div>
            <h3>{description}</h3>
            <p className="page-expense__data">{moment(createdAt).format('DD MMMM YYYY')}</p>
        </div>
        <p>{numeral(amount / 100).format('$0,0.00')}</p>
    </Link>
);

export default (ExpenseListItem); 