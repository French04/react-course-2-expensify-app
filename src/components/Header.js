import React from 'react';
import {NavLink} from 'react-router-dom';
import {startLogout} from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></p>
        <p><NavLink to="create" activeClassName="is-active">Create Expense</NavLink></p>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);