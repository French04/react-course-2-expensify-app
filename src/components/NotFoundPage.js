import React from 'react';
import {Link} from 'react-router-dom';

const NotFOundPage = () => (
    <div>
        404 ERROR! <Link to="/">Go home</Link>
    </div>
);

export default NotFOundPage;