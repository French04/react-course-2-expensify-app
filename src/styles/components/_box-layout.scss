.box-layout {
    align-items: center;                                //set component in center
    background: url('/images/bg.jpg');                  //set the background image from project directory folder
    background-size: cover;                             //set the image size to cover all background (justify method)
    display: flex;                                      //vertical box oriented
    height: 100vh;                                      //set box height to 100% of view port height
    justify-content: center;                            //set box in the midle
    width: 100vw;                                       //set box width to 100% of view port width
}

.box-layout__box {
    background: fade-out(white, .40);                 //create at semi-transparent backgraund on litle box (.50 is de percentual)
    border-radius: 3px;                                 //smooth the boarder angle
    padding: $large-size $medium-size;                  //space l-size from top and bottomo and m-sizo from left and right of little box
    text-align: center;                                 //set the text to center of litle box
    width: 25rem;                                       //set litle box width
}

.box-layout__title {
    margin: 0 0 $medium-size 0;
    line-height: 1;
}

// .box-layout__description {
//     font-weight: bold;
// }