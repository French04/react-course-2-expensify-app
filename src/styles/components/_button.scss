//Big button
.big-button {
    background: $orange;                               //colore dello sfondo del bottone
    border: none;                                      //nessun bordo al bottone
    border-radius: 3px;                                //smussa gli angoli al bottone
    border-bottom: .6rem solid darken($orange, 10%);   //regola la grandezza del bordo inferiore (.6rem) e lo scurisce del 10%
    color: white;                                    //colore del testo  
    font-weight: bold;                                 //scritta in bold
    font-size: $large-size;                            //grandezza del font
    margin-bottom: $medium-size;                       //spazio da lasciare sotto al bottone
    padding: 2.4rem;                                   //spazio tra la scritta e i margini sopra e sotto
    width: 100%;                                       //occupa tutto lo spazio del contenitore
}

//Regular Button
.regular-button {
    background: $orange;                               //colore dello sfondo del bottone
    border: none;                                      //nessun bordo al bottone
    border-radius: 3px;                                //smussa gli angoli al bottone
    border-bottom: .3rem solid darken($orange, 10%);   //regola la grandezza del bordo inferiore (.3rem) e lo scurisce del 10%
    color: white;                                    //colore del testo  
    font-weight: 300;                                  //scritta in bold500
    font-size: $font-size-large;                       //grandezza del font
    padding: $small-size;                              //spazio tra la scritta e i margini sopra e sotto
    display: inline-block;
    text-decoration: none;                             //remove link line under text
    line-height: 1;                                    //height of the text space
}

.remove-button {
    @extend .regular-button;
    background: $gray;
    border-bottom: .3rem solid darken($gray, 20%); 
}

.logout-button {
    background: none;                                  //remove the background on logout button
    border: none;                                      //remove the border
    color: white;                                    //text color to white
    font-weight: 100;                                  //text in bold 100
}