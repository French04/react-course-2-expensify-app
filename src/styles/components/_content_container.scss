.content_container {
    max-width: 80rem;                           //grandezza massima del contenitore
    margin: 0 auto;                             //allineamento margine a sinistra
    padding: 0 $medium-size;                    //spazio da lasciare dai margini sopra e sotto
}