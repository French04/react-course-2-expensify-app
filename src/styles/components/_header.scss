.header {
    background: $dark-orange;                       //set background color of header
}

.header__title {
    color: white;                                 //set title link color
    //text-decoration: none;                        //remove the line of link
    h1 {
        margin: 0;                                  //remove space from h1 margin
    }
}

.header__content {
    display:  flex;                                 //set the component on the same line
    align-items: center;                            //allign the component to the center of line
    justify-content: space-between;                 //set the space between the content components
    padding: $small-size 0;
}