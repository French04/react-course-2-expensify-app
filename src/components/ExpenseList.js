import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="page-expense__header">
            <div>Expense</div>
            <div>Amount</div>
        </div>
        <div className="page-expense_body">
            {
                props.expenses.length === 0 ? (
                    <div className="page-expense__item  page-expense__item--message">
                        <span>No Expenses</span>                
                    </div>
                ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} />;
                    })
                )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)        
    };
};

export default connect(mapStateToProps)(ExpenseList); 