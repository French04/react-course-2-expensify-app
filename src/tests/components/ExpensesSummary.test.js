import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('correctly render ExpensesSummary with 1 expense test', () => {
    const wrapper = shallow (<ExpensesSummary expenseCunt={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('correctly render ExpensesSummary with multiple expenses test', () => {
    const wrapper = shallow (<ExpensesSummary expenseCunt={23} expensesTotal={23512340987}/>);
    expect(wrapper).toMatchSnapshot();
});