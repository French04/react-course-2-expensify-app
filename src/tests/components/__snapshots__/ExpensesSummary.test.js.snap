// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly render ExpensesSummary with 1 expense test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCunt={1}
    expensesTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
         
        expenses
         totalling 
        $2.35
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          undefined,
          " ",
          "expenses",
          " totalling ",
          "$2.35",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        undefined,
        " ",
        "expenses",
        " totalling ",
        "$2.35",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
           
          expenses
           totalling 
          $2.35
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            undefined,
            " ",
            "expenses",
            " totalling ",
            "$2.35",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          undefined,
          " ",
          "expenses",
          " totalling ",
          "$2.35",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`correctly render ExpensesSummary with multiple expenses test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCunt={23}
    expensesTotal={23512340987}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
         
        expenses
         totalling 
        $235,123,409.87
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          undefined,
          " ",
          "expenses",
          " totalling ",
          "$235,123,409.87",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        undefined,
        " ",
        "expenses",
        " totalling ",
        "$235,123,409.87",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
           
          expenses
           totalling 
          $235,123,409.87
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            undefined,
            " ",
            "expenses",
            " totalling ",
            "$235,123,409.87",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          undefined,
          " ",
          "expenses",
          " totalling ",
          "$235,123,409.87",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
